### ISSUE TITLE
demonstrate using immutability in a reducer w/o Immutable (tm)

### ISSUE THREAD

**Issue by robwormald** [ngrx/angular2-store-example#12]
I'd like to see both approaches, for what its worth. No problem with using Immutable etc here, but i'd like it to be clear its not *required* to use ngrx

**Comment by fxck** [comment]
I agree. And also wonder if we should do the same for normalizr. Then again normalizr is relatively small library, unlike immutable.js.

**Comment by SekibOmazic** [comment]
If we want a simple app (just CRUD users) then we could either make a plunk and link it here or have another repo under ngrx (like angular2-store-simple). Without `Immutable.js`, `normalizr.js` et.al.

**Comment by meenie** [comment]
> If we want a simple app (just CRUD users) then we could either make a plunk and link it here or have another repo under ngrx (like angular2-store-simple). Without Immutable.js, normalizr.js et.al.

I like this idea.  I would rather not mix patterns in an app, even if it's just an example.  If this was a collection of small examples, then that would be cool.  But it would be confusing for people switching out the patterns in the different modules.

**Comment by btroncone** [comment]
I agree with Rob here, as someone who is picking up the library it was confusing at first to dive into the code and see all the normalizr/immutablejs specific functionality. I ended up spending as much time digging through those docs as I did trying to understand ngrx pattern and API. I think for the featured example it may be best to be Angular2/ngrx specific, as I feel that will be the most common use-case by far in the community. This could be maintained as a separate, more advanced example.

Just my two cents as someone who went through this the last few days. Keep up the great work!

**Comment by drew-moore** [comment]
@btroncone +1
